src/common/string_utils.hpp

```diff
@@ -0,0 +1,29 @@
+/* Copyright (C) 2021, Nikolai Wuttke. All rights reserved.
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <string>
+#include <vector>
+
+namespace rigel::strings {
+
+/** Split a string based on a single char delimiter and put the result into output
+ */
+void split(std::vector<std::string> &output, std::string const &input,
```

Comment:
I'd prefer returning the result, instead of using an out parameter. That would deviate more from the Boost API but I don't think that's an issue. Since there are not that many call sites it's easy to review the replacement.

Not sure why the Boost API was designed that way, maybe they didn't trust RVO back in the day? Or maybe it's for when you'd like to split multiple input strings into a single destination container? But that's not really a use case we have in RigelEngine right now, so we can go with a more straight-forward function signature.

Reply:
I think they had to work with a million compilers, some of which we might have never even heard about. And those might not do RVO properly.
So happy you want to use RVO. From now on I'll definitely do that first. The only reason I did it like that was for the compatibility-ish..
Will do